app/Domains/Contact/ManageContact/Dav/ExportNames.php

```diff
@@ -8,25 +8,36 @@
 use App\Models\Contact;
 use Sabre\VObject\Component\VCard;
 
+/**
+ * @implements ExportVCardResource<Contact>
+ */
 #[Order(1)]
 class ExportNames extends Exporter implements ExportVCardResource
 {
-    public function export(Contact $contact, VCard $vcard): void
+    public function getType(): string
+    {
+        return Contact::class;
+    }
+
+    /**
+     * @param  Contact  $resource
+     */
+    public function export($resource, VCard $vcard): void
```

Comment:
Shouldn't `$resource` be typed?

Reply:
I would like to type it `T`, but php doesn't accept it yet. It could however been type `mixed`, but it's worthless ...

Reply:
I've typed it `mixed` after all.